beets/dbcore/db.py

```diff
@@ -581,16 +581,6 @@ def add(self, db=None):
                     self._dirty.add(key)
             self.store()
 
-    # Formatting and templating.
-
-    _formatter = FormattedMapping
```

Comment:
This change is not as innocuous as it seems: The `_formatter` attribute is used by the `beets.library.Item` subclass of `Model` which replaces `FormattedMapping` by `FormattedItemMapping`. I would strongly suspect that this is the reason for the performance improvement, not having an additional factory method wrapping it. Without reading the code, I don't know what the exact implications are, but this commit must surely be breaking some of beets functionality.

There might of course be room for optimization in `FormattedItemMapping`.

Reply:
Ah, thanks! I couldn't figure out why this would impact performance so much. I guess I'll investigate this further.